docs/02-concepts/02-math/00-group.md

```diff
@@ -0,0 +1,71 @@
+- Associativity - $a + b + c = (a + b) + c = a + (b + c)$, where $a$, $b$ and $c$ are integers
+
+<br/>
+
+Conversely $(\mathbb{Z},*)$ or the set of integers under the multiplication operation is not a group:
+- $1$ is the identity element of $(\mathbb{Z},*)$
+- For an integer $a$, there is no inverse integer $a_{inv}$ such that $a * a_{inv} = 1$. For example given the integer $a = 2$, we cannot find another integer $a_{inv}$ such that $2*a_{inv} = 1$, unless we break out of the set of integers and grab a rational number $a_{inv} = \frac{1}{2}$
+
+### Order
+
+The **order** of a group is the number of elements in the group.
+
+The order of $(\mathbb{Z},+)$ or the set of integers under the addition is infinite. Under the hood in verkle tries, we make use of finite order groups.
+
+### Abelian group
+
+An abelian group (also called commutative group) is a group with an additional property of **commutativity** - $a+b = b+a$, where the order in which the the group operation is applied to two group elements does not alter the result. In verkle tries we deal exclusively with abelian groups under the hood.
+
+## Fields
+
+### Field definition
+
+A field F is a set of elements combined with two operations (commonly + and *) that can act on any two elements in the set. A field must satisfy the following properties:
+1. **Closure under both operations** - if $x, y \in F$, then $x + y$ and $x * y$ must also be elements of $F$
+2. **Identity elements for both operations** - $F$ contains identity elements $I_+$ and $I_*$ such that $a+I_+ = I_++a = a$, and $a*I_* = I_**a = a$
+3. **Inverse elements for both operations** - Every element $a \in F$, has inverses $a_+, a_* \in F$, such that $a+a_+ = I_+$ and $a*a_* = I_*$
+4. **Associativity under both operation** - $a+(b+c) = (a+b)+c$ and $a*(b*c) = (a*b)*c$
+5. **Commutativity under both operations** - $a+b = b+a$ and $a*b = b*a$
+6. **The second operation is distributive over the first operation** - $a*(b+c) = a*b+a*c$.
+
+Well known examples of fields are $(\mathbb{R}, +, *)$ - the set of real numbers under addition and multiplication, and $(\mathbb{Q}, +, *)$ - the set of rational numbers under addition and multiplication. Note that $(\mathbb{Z}, +, *)$ is not a field because there is no multiplicative inverse element.
+
+### Finite field
+
+A **finite field** (also called Galois field) is a field with a set containing a finite number of elements. An interesting mathematical result is that the number of elements (i.e. the order) in a finite field, must be a prime number $q$ or a prime power $q^n$.
+
+### Integers modulo prime - Fp
+
+The simplest type of finite fields are integers with modulo of a prime, denoted as $\mathbb{F}_p$ or $\mathbb{Z}/p\mathbb{Z}$ or $GF(p)$ (Galois field of p). For example $\mathbb{F}_{11}$ is {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10} and represents the set of all $a \bmod{11}$, where $a$ is any integer.
```

Comment:
integers with modulo of a prime -> integers modulo a prime number

Reads a bit better